$MirOS: ports/devel/cvs/patches/patch-diff_diff3_c,v 1.2 2007/03/10 23:53:47 tg Exp $
--- diff/diff3.c.orig	Sat Oct  9 15:51:30 2004
+++ diff/diff3.c	Sat Mar 10 23:14:14 2007
@@ -22,6 +22,8 @@
 #include "getopt.h"
 #include "diffrun.h"
 
+__RCSID("$MirOS: ports/devel/cvs/patches/patch-diff_diff3_c,v 1.2 2007/03/10 23:53:47 tg Exp $");
+
 /* diff3.c has a real initialize_main function. */
 #ifdef initialize_main
 #undef initialize_main
@@ -1439,12 +1441,12 @@ output_diff3 (diff, mapping, rev_mapping
      int const mapping[3], rev_mapping[3];
 {
   int i;
-  int oddoneout;
+  int oddoneout = 0;
   char *cp;
   struct diff3_block *ptr;
   int line;
   size_t length;
-  int dontprint;
+  int dontprint = 0;
   static int skew_increment[3] = { 2, 3, 1 }; /* 0==>2==>1==>3 */
   char const *line_prefix = tab_align_flag ? "\t" : "  ";
 
@@ -1555,11 +1557,12 @@ undotlines (leading_dot, start, num)
      int leading_dot, start, num;
 {
   write_output (".\n", 2);
-  if (leading_dot)
+  if (leading_dot) {
     if (num == 1)
       printf_output ("%ds/^\\.//\n", start);
     else
       printf_output ("%d,%ds/^\\.//\n", start, start + num - 1);
+  }
 }
 
 /*
@@ -1750,11 +1753,12 @@ output_diff3_merge (infile, diff, mappin
 	do
 	  {
 	    c = getc (infile);
-	    if (c == EOF)
+	    if (c == EOF) {
 	      if (ferror (infile))
 		diff3_perror_with_exit ("input file");
 	      else if (feof (infile))
 		diff3_fatal ("input file shrank");
+	    }
 	    cc = c;
 	    write_output (&cc, 1);
 	  }
@@ -1804,7 +1808,7 @@ output_diff3_merge (infile, diff, mappin
       linesread += i;
       while (0 <= --i)
 	while ((c = getc (infile)) != '\n')
-	  if (c == EOF)
+	  if (c == EOF) {
 	    if (ferror (infile))
 	      diff3_perror_with_exit ("input file");
 	    else if (feof (infile))
@@ -1813,6 +1817,7 @@ output_diff3_merge (infile, diff, mappin
 		  diff3_fatal ("input file shrank");
 		return conflicts_found;
 	      }
+	  }
     }
   /* Copy rest of common file.  */
   while ((c = getc (infile)) != EOF || !(ferror (infile) | feof (infile)))
